lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -136,6 +128,54 @@ def build_client(options)
       Elasticsearch::Client.new(client_options)
     end
 
+    HOSTNAME_PORT_REGEX=/\A(?<hostname>([A-Za-z0-9\.\-]+))(:(?<port>\d+))?\Z/
+    # Parse a configuration host to a normalized URL
+    def host_to_url(host, ssl=nil, path=nil)
+      explicit_scheme = if ssl.nil?
+                          nil
+                        else
+                          ssl ? "https"  : "http"
+                        end
+
+      # Ensure path starts with a /
+      if path && path[0] != '/'
+        path = "/#{path}"
+      end
+
+      url = nil
+      if host =~ /\A#{URI::regexp(['http', 'https'])}\z/
+        url = URI.parse(host)
+        if url.scheme == "http" && ssl == true
```

Comment:
style: instead of `ssl == true` and later `ssl == false` -- thoughts on just using `ssl` as the value?

like: `url.scheme == "http" && ssl`